### ISSUE TITLE
issue with http get

### ISSUE THREAD

**Issue by rcanetta** [Nekmo/amazon-dash#104]
Put an `x` into all the boxes [ ] relevant to your *issue* (like this: `[x]`)

### What is the purpose of your *issue*?
- [ ] Bug report (encountered problems with amazon-dash)
- [ ] Feature request (request for a new functionality)
- [x] Question
- [ ] Other

#### Description

Hi
I'm trying to configure dash button to perform a get with authentication

```
EE:08:6B:21:CC:C3:
    name: Daygum 
    url: 'http://10.16.3.250/api/sceneControl'  # Url to execute
    method: get  # HTTP method. By default GET
    headers: {"authorization": "'myusername','mypassword'"}
#    content-type: json  # Only available if Body is defined
   body: '{"mac": "EE:08:6B:21:CC:C3","action":"start","id":"10"}'  # Request payload. Remember the quotes
```
with check-config I am getting error on the body line. Any idea of what am I doing wrong?

Thanks!
R

**Comment by Nekmo** [comment]
Hi, your device must be well indented.

```
EE:08:6B:21:CC:C3:
    name: Daygum 
    url: 'http://10.16.3.250/api/sceneControl'  # Url to execute
    method: get  # HTTP method. By default GET
    headers: {"authorization": "'myusername','mypassword'"}
#    content-type: json  # Only available if Body is defined
    body: '{"mac": "EE:08:6B:21:CC:C3","action":"start","id":"10"}'  # Request payload. Remember the quotes
```

Instead:
```
EE:08:6B:21:CC:C3:
    name: Daygum 
    url: 'http://10.16.3.250/api/sceneControl'  # Url to execute
    method: get  # HTTP method. By default GET
    headers: {"authorization": "'myusername','mypassword'"}
#    content-type: json  # Only available if Body is defined
   body: '{"mac": "EE:08:6B:21:CC:C3","action":"start","id":"10"}'  # Request payload. Remember the quotes
```

You can check that "body" is not well indented.

regards.

**Comment by rcanetta** [comment]
thanks I fixed that but now I get error 401. I tried changing the authorization line like this
`  EE:08:6B:21:CC:C3:
    name: Daygum
    url: 'http://10.16.3.250/api/sceneControl'  # Url to execute
    method: put  # HTTP method. By default GET
    headers: {"authorization": "username:password"}
#    content-type: json  # Only available if Body is defined
    body: '{"mac": "EE:08:6B:21:CC:C3","action":"start","id":"10"}'  # Request payload. Remember the quotes
`
but I still get 401...

Any hint on how to configure username/password?

Thanks!
R

**Comment by Nekmo** [comment]
Are you sure that your service requires logging as you are doing? Headers usually start with a capital letter. For example: Authorization.

**Comment by rcanetta** [comment]
Hi I need to authenticate the session, yes - but I don't know how to do it.. I tried changing authorization to Authorization but I still get 401...

Thanks
R

**Comment by rcanetta** [comment]
Hi, I solved by creating a python script. Now when I run the script manually ("./home/pi/allarme") as user "pi" it works ok but when it's triggered from
`
  ee:08:6b:21:cc:c3:
    name: Daygum
    user: pi
    cmd: /home/pi/allarme
`
I still get error 401... I tried creating the script from scratch as user "pi" and then giving chmod 755 , but I still get this error..

Any idea of what could be wrong?

Thanks!
R
